microcosm_pubsub/daemon.py

```diff
@@ -16,34 +14,36 @@ def make_arg_parser(self):
         parser.add_argument("--sqs-queue-url")
         return parser
 
-    @abstractproperty
     def schema_mappings(self):
         """
         Define the PubSub message media-type to schema mappings.
 
         """
-        pass
+        return dict()
+
+    def create_object_graph(self, args):
+        graph = super(ConsumerDaemon, self).create_object_graph(args)
+        for media_type, schema_cls in self.schema_mappings.items():
+            self.graph.pubsub_message_schema_registry.register(media_type, schema_cls)
+        return graph
 
     @property
     def defaults(self):
-        dct = dict(
-            pubsub_message_codecs=dict(
```

Comment:
No, `pubsub_message_codecs` is a translation between `media_type` and `schema`. It's used for both publish and subscribe. Part of why this PR is valuable is exactly this confusion; we frequently had handlers get registered for a `media_type`, but not the accompanying `schema`; this meant that we never decoded the message to pass to the schema and the daemon didn't work. The new `@schema` decorator makes that problem vanish by binding the handler to a schema class.